tests/otel/index.test.ts

```diff
@@ -0,0 +1,63 @@
+import { describe, it } from 'node:test';
+import * as assert from 'node:assert';
+import * as yaml from 'yaml';
+import { OtelCollectorConfigBuilder } from '../../src/v2/otel/config';
+
+const aws = { config: { region: 'us-west-2' } };
+const prometheusNamespace = 'test-namespace';
+const prometheusWriteEndpoint = 'https://aps-workspaces.us-west-2.amazonaws.com/workspaces/ws-12345/api/v1/remote_write';
+
+describe('OtelCollectorConfigBuilder', () => {
+  it('should generate correct configuration', () => {
+    const yamlOutput = new OtelCollectorConfigBuilder()
+      .withOTLPReceiver(['http'])
+      .withBatchProcessor()
+      .withMemoryLimiterProcessor()
+      .withAPS(prometheusNamespace, prometheusWriteEndpoint, aws.config.region)
+      .withAWSXRayExporter(aws.config.region)
+      .withDebug()
+      .withHealthCheckExtension()
+      .withPprofExtension()
+      .withMetricsPipeline(
+        ["otlp"],
+        ["memory_limiter", "batch"],
+        ["prometheusremotewrite", "debug"]
+      )
+      .withTracesPipeline(
+        ["otlp"],
+        ["memory_limiter", "batch"],
+        ["awsxray", "debug"]
+      )
+      .build();
+
+    const result = yaml.parse(yamlOutput);
+
+    assert.deepStrictEqual(result.receivers.otlp.protocols.http, { endpoint: "0.0.0.0:4318" });
```

Comment:
This seems as too many asserts to have in one test, and it seems they all are asserting different properties of the same object. 

Why not just test the whole objects using `deepStrictEqual`? It should improve readability.

Reply:
When there's a problem, it becomes much harder to see what went wrong based on the diff, as it's big.
I'll split this test to many smaller tests. Should make them easier to read, extend and determine where the issue is when they break.